content/automate/ManageIQ/PhysicalInfrastructure/CiscoIntersight/GetMethods/GetServers.class/__methods__/get_servers.rb

```diff
@@ -0,0 +1,25 @@
+# Description: get unassigned servers
+$evm.log("info", 'selected_ems for servers')
+
+dialog_field = $evm.object
+dialog_field["sort_by"] = "value"
+dialog_field["sort_order"] = "ascending"
+dialog_field["data_type"] = "string"
+dialog_field["required"] = true
+
+selected_ems_id = $evm.root.attributes['dialog_ems']
+selected_ems = $evm.vmdb(:ext_management_system).where(:id=>selected_ems_id).first
+$evm.log("info", 'selected_ems for servers')
+$evm.log("info", selected_ems.name)
+
+servers = selected_ems.physical_servers
+profiles = selected_ems.physical_server_profiles
+profiles_with_server = profiles.reject { |p| p["assigned_server_id"].nil? }
+servers_with_profile = profiles_with_server.pluck("assigned_server_id")
```

Comment:
This is a question because I don't know automate well, but can you `selected_ems.physical_server_profiles.where.not(:assigned_server_id => nil).pluck(:assigned_server_id)` like we would in normal rails code?

Reply:
selected_ems.physical_server_profiles return an array so I can't create a query on it. 
And also we can't query the PhysicalServerProfileTemplate model from the automation engine as far as I know...